### DISCUSSION TITLE
Err: only integer scalar arrays can be converted to a scalar index!

### DISCUSSION THREAD

**Discussion by LiuZhihhxx** [PaddlePaddle/PaddleSpatial#183] [BaiduKDDCup2022] [answered]
I received an error message "only integer scalar arrays can be converted to a scalar index"(following picture) but not clear what it means.
![image](https://user-images.githubusercontent.com/73637604/174245528-86beb102-932b-4176-905f-9ead3cf6d27e.png)

I tried to dubug on my local machine by establishing an environment like "Backend Files for Evaluation"(https://github.com/PaddlePaddle/PaddleSpatial/tree/main/apps/wpf_baseline_gru/kddcup22-sdwpf-evaluation)

However another error occured: 
![image](https://user-images.githubusercontent.com/73637604/174246368-08363e64-d110-4174-8b1e-7cc1e86afb94.png)
I've got it through that it's because NUM_MAX_RUNS = 196 (defalt setting in evaluation.py) but only 1 pair of "test_x" and "test_y" in the environment. And when I set NUM_MAX_RUNS = 1 and ignored the std error(line 273-278 in evaluation.py), everything went well and it yielded final MAE, RMSE, and Final Score.

I'm confused that:
(1) What lead to the first error?
(2) Why different errors occured online and locally?
(3) To locally debug, is it necessary to prepare 196 pairs of test_x and test_y in the local environment?

Any help will be appreciated.

Team CSU123, last submission: 2022-06-17 15:01

**Comment by ramber1836** [comment, marked as answer]
This  is the traceback log:

Traceback (most recent call last):
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/paddlepaddle/evaluation.py", line 332, in eval
    return evaluate(tmp_dir)
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/paddlepaddle/evaluation.py", line 233, in evaluate
    pred_res = predict_and_test(envs, PRED_DIR, forecast_module, i, flag='predict')
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/paddlepaddle/evaluation.py", line 168, in predict_and_test
    return exec_predict_and_test(envs, path_to_test_file, forecast_module, flag)
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/paddlepaddle/evaluation.py", line 131, in exec_predict_and_test
    "prediction": forecast_module.forecast(envs)
  File "/tmp/tmpyykaj9qs/predict.py", line 94, in forecast
TypeError: only integer scalar arrays can be converted to a scalar index
